### PR TITLE
Get build working on Xcode 16

### PR DESCRIPTION
**Pull request by edgchen1**
### Description

Get build working on Xcode 16.

### Motivation and Context

To get ready for more usage of Xcode 16.

### CODE REVIEW

File: .github/workflows/mac.yml
Comment (edgchen1) [inline .github/workflows/mac.yml:1]: ignore whitespace changes when viewing this file. the line endings got converted from CRLF to LF.

File: cmake/patches/nsync/nsync_1.26.0.patch [hunk 1]
```diff
@@ -0,0 +1,14 @@
+diff --git a/public/nsync_atomic.h b/public/nsync_atomic.h
```
Comment (edgchen1) [inline cmake/patches/nsync/nsync_1.26.0.patch:1]: FYI, nsync PR: https://github.com/google/nsync/pull/23

File: onnxruntime/test/optimizer/optimizer_test.cc [hunk 2]
```diff
@@ -24,8 +24,6 @@ using namespace ONNX_NAMESPACE;
 namespace onnxruntime {
 namespace test {
 
-static const std::string MODEL_FOLDER = "testdata/transform/";
```
Comment (edgchen1) [inline onnxruntime/test/optimizer/optimizer_test.cc:27]: this was unused

**Review by snnn — APPROVED** [review]

**Review by skottmckay — APPROVED** [review]

**Pull request merged**